lib/app/features/config/providers/config_provider.c.dart

```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: ice License 1.0
+
+import 'dart:io';
+
+import 'package:dio/dio.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:ion/app/exceptions/exceptions.dart';
+import 'package:ion/app/features/core/providers/dio_provider.c.dart';
+import 'package:ion/app/features/core/providers/env_provider.c.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+part 'config_provider.c.g.dart';
+
+@riverpod
+ConfigService configService(Ref ref) => ConfigService(ref.watch(dioProvider));
+
+class ConfigService {
```

Comment:
do we need this class? can you consider creating a functional provider instead of ConfigService class?